diff --git a/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-plugin/src/main/kotlin/expo/modules/plugin/ExpoRootProjectPlugin.kt b/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-plugin/src/main/kotlin/expo/modules/plugin/ExpoRootProjectPlugin.kt
index 45b22dc..a2e9a13 100644
--- a/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-plugin/src/main/kotlin/expo/modules/plugin/ExpoRootProjectPlugin.kt
+++ b/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-plugin/src/main/kotlin/expo/modules/plugin/ExpoRootProjectPlugin.kt
@@ -8,7 +8,6 @@ import org.gradle.api.Project
 import org.gradle.api.artifacts.VersionCatalog
 import org.gradle.api.artifacts.VersionCatalogsExtension
 import org.gradle.api.plugins.ExtraPropertiesExtension
-import org.gradle.internal.extensions.core.extra
 import java.util.Optional
 import kotlin.jvm.optionals.getOrNull
 
@@ -48,6 +47,7 @@ private fun Project.maybeOverrideCmakeVersion() {
 }
 
 fun Project.defineDefaultProperties(versionCatalogs: Optional<VersionCatalog>) {
+  val extra = extensions.extraProperties
   // Android related
   val buildTools = extra.setIfNotExist("buildToolsVersion") { versionCatalogs.getVersionOrDefault("buildTools", "35.0.0") }
   val minSdk = extra.setIfNotExist("minSdkVersion") { Integer.parseInt(versionCatalogs.getVersionOrDefault("minSdk", "24")) }
diff --git a/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-settings-plugin/src/main/kotlin/expo/modules/plugin/ExpoAutolinkingSettingsPlugin.kt b/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-settings-plugin/src/main/kotlin/expo/modules/plugin/ExpoAutolinkingSettingsPlugin.kt
index d66e371..08a0591 100644
--- a/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-settings-plugin/src/main/kotlin/expo/modules/plugin/ExpoAutolinkingSettingsPlugin.kt
+++ b/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-settings-plugin/src/main/kotlin/expo/modules/plugin/ExpoAutolinkingSettingsPlugin.kt
@@ -9,7 +9,7 @@ import expo.modules.plugin.utils.getPropertiesPrefixedBy
 import org.gradle.api.Plugin
 import org.gradle.api.UnknownProjectException
 import org.gradle.api.initialization.Settings
-import org.gradle.internal.cc.base.logger
+import org.gradle.api.logging.Logging
 import java.io.File
 import java.util.Properties
 
@@ -71,7 +71,7 @@ open class ExpoAutolinkingSettingsPlugin : Plugin<Settings> {
           }
         }
       } catch (e: UnknownProjectException) {
-        logger.error(
+        Logging.getLogger(ExpoAutolinkingSettingsPlugin::class.java).error(
           " ℹ️  Failed to apply gradle plugin ".withColor(Colors.RESET)
             + "'expo-max-sdk-override-plugin'".withColor(Colors.GREEN)
             + ". Plugin has failed to find the ':app' project. It will not be applied.".withColor(Colors.RESET)
diff --git a/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-settings-plugin/src/main/kotlin/expo/modules/plugin/SettingsManager.kt b/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-settings-plugin/src/main/kotlin/expo/modules/plugin/SettingsManager.kt
index 98ee3dd..6c2fcc2 100644
--- a/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-settings-plugin/src/main/kotlin/expo/modules/plugin/SettingsManager.kt
+++ b/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-autolinking-settings-plugin/src/main/kotlin/expo/modules/plugin/SettingsManager.kt
@@ -21,7 +21,6 @@ import groovy.lang.GroovyShell
 import org.gradle.api.Project
 import org.gradle.api.initialization.Settings
 import org.gradle.api.logging.Logging
-import org.gradle.internal.extensions.core.extra
 import java.io.File
 
 class SettingsManager(
@@ -107,7 +106,7 @@ class SettingsManager(
 
     // Defines the required features for the core module
     settings.gradle.beforeProject("expo-modules-core") { project ->
-      project.extra.set("coreFeatures", config.coreFeatures)
+      project.extensions.extraProperties.set("coreFeatures", config.coreFeatures)
     }
 
     settings.gradle.beforeRootProject { rootProject: Project ->
diff --git a/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-max-sdk-override-plugin/src/main/kotlin/expo/modules/plugin/ExpoMaxSdkOverridePlugin.kt b/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-max-sdk-override-plugin/src/main/kotlin/expo/modules/plugin/ExpoMaxSdkOverridePlugin.kt
index c90363f..b7e8b66 100644
--- a/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-max-sdk-override-plugin/src/main/kotlin/expo/modules/plugin/ExpoMaxSdkOverridePlugin.kt
+++ b/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-max-sdk-override-plugin/src/main/kotlin/expo/modules/plugin/ExpoMaxSdkOverridePlugin.kt
@@ -7,19 +7,11 @@ import com.android.build.api.variant.ApplicationAndroidComponentsExtension
 import expo.modules.plugin.text.Colors
 import expo.modules.plugin.text.Emojis
 import expo.modules.plugin.text.withColor
-import org.gradle.internal.cc.base.logger
-
-/**
- * Plugin, which registers ExpoMaxSdkOverrideTask and schedules it to run with `app:processDebugManifest`.
- *
- * The task finds all permissions declared with `android:maxSdkVersion`. If the permission was declared in more than one place, and one of the places
- * defines the task without `android:maxSdkVersion` the task will remove the `android:maxSdkVersion` from the final merged manifest
- */
 class ExpoMaxSdkOverridePlugin : Plugin<Project> {
   override fun apply(project: Project) {
     val androidComponents = project.extensions.getByType(ApplicationAndroidComponentsExtension::class.java)
-    logger.quiet(" ${Emojis.INFORMATION}  ${"Applying gradle plugin".withColor(Colors.YELLOW)} '${"expo-max-sdk-override-plugin".withColor(Colors.GREEN)}'")
-    logger.quiet("  [expo-max-sdk-override-plugin] This plugin will find all permissions declared with `android:maxSdkVersion`. If there exists a declaration with the `android:maxSdkVersion` annotation and another one without, the plugin will remove the annotation from the final merged manifest. In order to see a log with the changes run a clean build of the app.")
+    project.logger.quiet(" ${Emojis.INFORMATION}  ${"Applying gradle plugin".withColor(Colors.YELLOW)} '${"expo-max-sdk-override-plugin".withColor(Colors.GREEN)}'")
+    project.logger.quiet("  [expo-max-sdk-override-plugin] This plugin will find all permissions declared with `android:maxSdkVersion`. If there exists a declaration with the `android:maxSdkVersion` annotation and another one without, the plugin will remove the annotation from the final merged manifest. In order to see a log with the changes run a clean build of the app.")
 
     androidComponents.onVariants(androidComponents.selector().all()) { variant ->
       val taskName = "expo${variant.name.replaceFirstChar { it.uppercase() }}OverrideMaxSdkConflicts"
diff --git a/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-max-sdk-override-plugin/src/main/kotlin/expo/modules/plugin/FindPermissionsToOverride.kt b/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-max-sdk-override-plugin/src/main/kotlin/expo/modules/plugin/FindPermissionsToOverride.kt
index f47d28b..b2f0cc2 100644
--- a/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-max-sdk-override-plugin/src/main/kotlin/expo/modules/plugin/FindPermissionsToOverride.kt
+++ b/node_modules/expo-modules-autolinking/android/expo-gradle-plugin/expo-max-sdk-override-plugin/src/main/kotlin/expo/modules/plugin/FindPermissionsToOverride.kt
@@ -1,6 +1,6 @@
 package expo.modules.plugin
 
-import org.gradle.internal.cc.base.logger
+import org.gradle.api.logging.Logging
 import java.io.File
 import javax.xml.parsers.DocumentBuilderFactory
 
@@ -11,6 +11,7 @@ import javax.xml.parsers.DocumentBuilderFactory
  * @param problematicPermissions A Map of `String` and `PermissionInfo` obtained with analyzeManifestReport
  */
 internal fun findPermissionsToOverride(problematicPermissions: Map<String, PermissionInfo>): Map<String, PermissionInfo> {
+  val logger = Logging.getLogger("FindPermissionsToOverride")
   val factory = DocumentBuilderFactory.newInstance()
   factory.isNamespaceAware = true
 
